### DISCUSSION TITLE
[1.1.438] SOTA SoftEdge TEED

### DISCUSSION THREAD

**Discussion by sdbds** [Mikubill/sd-webui-controlnet#2575] [General]
## TEED
https://github.com/xavysp/TEED

## How to use
SD1.5: 
control_v11p_sd15_softedge
SDXL:
Download SargeZT/controlnet-sd-xl-1.0-softedge-dexined [here](https://huggingface.co/SargeZT/controlnet-sd-xl-1.0-softedge-dexined).

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/8085926/0d343df3-5616-4155-8b4a-9d4f23144697)
Default with `safe steps=2`, you can set `safe steps=0` to get original repo effect.  

## Preprocessor Comparison
|input| hed | pidinet | **TEED** | Lineart anime| Lineart realistic
|--------|--------|--------|--------|--------|--------|
|![image](https://github.com/Mikubill/sd-webui-controlnet/assets/8085926/b97ce5e8-ebe3-4d95-859f-9cdd86a58c4a)|![image](https://github.com/Mikubill/sd-webui-controlnet/assets/8085926/e7c3f534-3cb8-4179-8774-57dba2b9b170)|![image](https://github.com/Mikubill/sd-webui-controlnet/assets/8085926/f09ca04b-69e3-418a-8771-4f6c6f6797e6)|![image](https://github.com/Mikubill/sd-webui-controlnet/assets/8085926/3baf838f-ecfd-4dfc-9bf1-c084856a5f4e)|![image](https://github.com/Mikubill/sd-webui-controlnet/assets/8085926/af674522-2852-4be0-be79-4f53b0be449a)|![image](https://github.com/Mikubill/sd-webui-controlnet/assets/8085926/6d1a7477-6e19-4179-8c87-cdef1020d450)

## Introduction
#2093 

I would like to introduce a friend, SargeZT(Patrick Erin Shanahan), who was not very familiar to me but worked diligently in SDXL. SargeZT's sudden passing has taken away a passionate developer from us.

As a developer, SargeZT devoted a great deal of passion and effort to building SDXL controlnet models. His love for technology and dedication to his projects were inspiring to us.

Even though we only talked a few times, I was deeply impressed by his focus and enthusiasm. To honor SargeZT, I have decided to take over his unfinished projects so update this preprocesser. This is not only a continuation of his work but also a tribute to his spirit.

Here, I express my respect to SargeZT. His work and passion will inspire us to keep moving forward.

Thank you for reading, and let us collectively commemorate SargeZT, remembering his enthusiasm and contributions to controlnet community.

## Conslusion
TEED had more details than other line preprocessor.

**Comment by if-ai** [comment]
Sad to hear his brilliant mind is not with us anymore. Rest in peace SargeZT. Thank you for continuing up on his legacy efforts😢

**Comment by IndigoDosSantos** [comment]
This is great, beautiful words! :heart:

**Comment by IndigoDosSantos** [comment]
Very promising, especially when you consider that no prompt, but only controlnet-sd-xl-1.0-softedge-dexined together with teed preprocessor, was used.

![xyz_grid-0002-1953010459_psMerge](https://github.com/Mikubill/sd-webui-controlnet/assets/130935112/c0643c11-c700-43c0-aa3a-1ff67ceac75b)
